octopusdeploy/schema_kubernetes_agent_worker.go

```diff
@@ -0,0 +1,156 @@
+package octopusdeploy
+
+import (
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/machines"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"net/url"
+)
+
+func expandKubernetesAgentWorker(kubernetesAgent *schema.ResourceData) *machines.Worker {
+	uri, _ := url.Parse(kubernetesAgent.Get("uri").(string))
+	thumbprint := kubernetesAgent.Get("thumbprint").(string)
+
+	communicationsMode := kubernetesAgent.Get("communication_mode").(string)
+	upgradeLocked := kubernetesAgent.Get("upgrade_locked").(bool)
+	var endpoint machines.IEndpoint = machines.NewKubernetesTentacleEndpoint(uri, thumbprint, upgradeLocked, communicationsMode, "")
+
+	name := kubernetesAgent.Get("name").(string)
+	Worker := machines.NewWorker(name, endpoint)
+
+	Worker.IsDisabled = kubernetesAgent.Get("is_disabled").(bool)
+	Worker.Thumbprint = thumbprint
+	Worker.WorkerPoolIDs = getSliceFromTerraformTypeList(kubernetesAgent.Get("worker_pool_ids"))
+
+	Worker.SpaceID = kubernetesAgent.Get("space_id").(string)
+
+	return Worker
+}
+
+func flattenKubernetesAgentWorker(Worker *machines.Worker) map[string]interface{} {
+	if Worker == nil {
+		return nil
+	}
+
+	if Worker.Endpoint.GetCommunicationStyle() != "KubernetesTentacle" {
+		return nil
+	}
+
+	endpoint := Worker.Endpoint.(*machines.KubernetesTentacleEndpoint)
+
+	flattenedWorker := map[string]interface{}{}
+	flattenedWorker["id"] = Worker.GetID()
+	flattenedWorker["space_id"] = Worker.SpaceID
+	flattenedWorker["name"] = Worker.Name
+	flattenedWorker["machine_policy_id"] = Worker.MachinePolicyID
+	flattenedWorker["is_disabled"] = Worker.IsDisabled
+
+	flattenedWorker["thumbprint"] = endpoint.TentacleEndpointConfiguration.Thumbprint
+	flattenedWorker["uri"] = endpoint.TentacleEndpointConfiguration.URI.String()
+	flattenedWorker["communication_mode"] = endpoint.TentacleEndpointConfiguration.CommunicationMode
+	flattenedWorker["worker_pool_ids"] = Worker.WorkerPoolIDs
+
+	if endpoint.KubernetesAgentDetails != nil {
+		flattenedWorker["agent_version"] = endpoint.KubernetesAgentDetails.AgentVersion
+		flattenedWorker["agent_tentacle_version"] = endpoint.KubernetesAgentDetails.TentacleVersion
+		flattenedWorker["agent_upgrade_status"] = endpoint.KubernetesAgentDetails.UpgradeStatus
+		flattenedWorker["agent_helm_release_name"] = endpoint.KubernetesAgentDetails.HelmReleaseName
+		flattenedWorker["agent_kubernetes_namespace"] = endpoint.KubernetesAgentDetails.KubernetesNamespace
+	}
+
+	return flattenedWorker
+}
+
+func getKubernetesAgentWorkerSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"id":       getIDSchema(),
+		"space_id": getSpaceIDSchema(),
+		"name":     getNameSchema(true),
+		"communication_mode": {
```

Comment:
Nit: Can we just exclude this field until there is support for more than one communication mode? We are going to have to come back and edit the description once that happens anyway

Reply:
In an attempt to keep it aligned with the k8s_deployment_target, I was going to leave it here.